### PR TITLE
Refactoring navigation menu component

### PR DESCRIPTION
**Pull request by andriifromlviv**
removed unnecessary items from NavigationMenuPageComponent, added components for authorized and unauthorized users, create subfolder for NavigationMenu components

### CODE REVIEW

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/UserMenuDropDownListPageComponent.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+﻿using System;
```
Comment (dmatsev) [inline TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/UserMenuDropDownListPageComponent.cs:1]: please cleanup namespaces

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+﻿using System;
```
Comment (dmatsev) [inline TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs:1]: please cleanup

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/NavigationMenuPageComponent.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+﻿namespace TestFramework.PageComponents.NavigationMenuComponents
+{
+    public class NavigationMenuPageComponent : BasePage
+    {
+        public NavigationMenuPageComponent(IWebDriver driver) : base(driver)
+        {
+            wait = new WebDriverWait(driver, timeout);
+        }
+
+        protected static TimeSpan timeout = TimeSpan.FromSeconds(3);
```
Comment (dmatsev) [inline TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/NavigationMenuPageComponent.cs:10]: This is what is supposed to be on the test level, page object doesn't know what timeout should be for searching elements.

**Review by dmatsev — CHANGES_REQUESTED** [review]
Please resolve the comments

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/NavigationMenuPageComponent.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+﻿namespace TestFramework.PageComponents.NavigationMenuComponents
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/NavigationMenuPageComponent.cs:1]: image
This must be the same name.

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs:6]: Remove lines 1-5

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+namespace TestFramework.PageComponents.NavigationMenuComponents
+{
+    public class SignInSignUpHeaderButtons
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs:9]: Add PageComponents in the end of the name.

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        IWebDriver driver { get; set; }
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs:11]: Do we need a setter here?

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+        public SignInSignUpHeaderButtons(IWebDriver driver)
+        {
+            this.driver = driver;
+        }
+
+        private IWebElement _SignIn => driver.FindElement(By.XPath("//span[text()='Sign In']"));
+        private IWebElement _SignUp => driver.FindElement(By.XPath("//span[text()='Sign Up']"));
+
+        public SignInPage ClickOnSignInButton()
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs:21]: ClickSignInButton instead.

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            _SignIn.Click();
+            return new SignInPage(driver);
+        }
+
+        public SignUpPage ClickOnSignUpButton()
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs:27]: ClickSignUpButton instead.

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/UserMenuDropDownListPageComponent.cs [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        public UserMenuDropDownListPageComponent(IWebDriver driver)
+        {
+            wait = new WebDriverWait(driver, timeout);
+            this.driver = driver;
+        }
+
+        private IWebElement _MyProfile = wait.Until(drv => drv.FindElement(By.XPath("//a[@role='menuitem']")));
+        private IWebElement _LogOut = wait.Until(drv => drv.FindElement(By.XPath("//li[text()='Log Out']")));
+
+        public void ClickGoToMyProfileButton()
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/UserMenuDropDownListPageComponent.cs:25]: ClickMyProfileButton instead.

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/UserMenuHeaderButton.cs [hunk 11]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TestFramework.PageComponents.NavigationMenuComponents
+{
+    public class UserMenuHeaderButton
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/UserMenuHeaderButton.cs:9]: Add PageComponent in the end

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/UserMenuHeaderButton.cs [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+        IWebDriver driver { get; set; }
+
+        protected static TimeSpan timeout = TimeSpan.FromSeconds(3);
+
+        public UserMenuHeaderButton(IWebDriver driver)
+        {
+            this.driver = driver;
+        }
+
+        private IWebElement _UserMenu => driver.FindElement(By.XPath("//button/span[@class='MuiIconButton-label-2477']/div"));
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/UserMenuHeaderButton.cs:20]: The name must start in lowercase.

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/UserMenuHeaderButton.cs [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+
+        public UserMenuDropDownListPageComponent ClickOnUserMenuButton()
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/UserMenuHeaderButton.cs:22]: ClickUserMenuButton instead

File: TAF_EasyRest/TestFramework/Pages/HomePageNonAuthorizedUser.cs [hunk 14]
```diff
@@ -0,0 +1,16 @@
+﻿using TestFramework.PageComponents.NavigationMenuComponents;
+
+namespace TestFramework.Pages
+{
+    public class HomePageNonAuthorizedUser : BasePage
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/Pages/HomePageNonAuthorizedUser.cs:5]: Add Page in the end.

**Review by LoveMyDaughter — CHANGES_REQUESTED** [review]
Please make changes.

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/NavigationMenuPageComponent.cs [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+            wait = new WebDriverWait(driver, timeout);
+        }
+
+        protected static TimeSpan timeout = TimeSpan.FromSeconds(3);
+        protected static WebDriverWait wait;
+
+        private IWebElement _HomeButton => driver.FindElement(By.XPath("//span[text()='Home']"));
+        private IWebElement _RestaurantsList => driver.FindElement(By.XPath("//span[text()='Restaurants List']"));
+
+        public HomePageNonAuthorizedUser ClickHomeButton()
```
[the remark this answers is not in the document]
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/NavigationMenuPageComponent.cs:16]: changed

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/NavigationMenuPageComponent.cs [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+        {
+            _HomeButton.Click();
+            return new HomePageNonAuthorizedUser(driver);
+        }
+
+        public RestaurantsListPage ClickRestaurantsListButton()
+        {
+            _RestaurantsList.Click();
```
[the remark this answers is not in the document]
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/NavigationMenuPageComponent.cs:24]: changed this and other names according to this practice

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/NavigationMenuPageComponent.cs [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        public NavigationMenuPageComponent(IWebDriver driver) : base(driver)
+        {
+            wait = new WebDriverWait(driver, timeout);
+        }
+
+        protected static TimeSpan timeout = TimeSpan.FromSeconds(3);
+        protected static WebDriverWait wait;
+
+        private IWebElement _HomeButton => driver.FindElement(By.XPath("//span[text()='Home']"));
```
[the remark this answers is not in the document]
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/NavigationMenuPageComponent.cs:13]: maybe, we should delete WebDriverWait class instances at all from our pageobjects and add them to tests

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/NavigationMenuPageComponent.cs [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+        private IWebElement _RestaurantsList => driver.FindElement(By.XPath("//span[text()='Restaurants List']"));
```
[the remark this answers is not in the document]
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/NavigationMenuPageComponent.cs:14]: changed

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/NavigationMenuPageComponent.cs (hunk 4 above)
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/NavigationMenuPageComponent.cs:1]: changed

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs (hunk 5 above)
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs:6]: done

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs (hunk 6 above)
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs:9]: added

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs (hunk 7 above)
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs:11]: I guess no, removed

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/UserMenuDropDownListPageComponent.cs (hunk 10 above)
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/UserMenuDropDownListPageComponent.cs:25]: changed

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/UserMenuHeaderButton.cs (hunk 11 above)
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/UserMenuHeaderButton.cs:9]: added

File: TAF_EasyRest/TestFramework/Pages/HomePageNonAuthorizedUser.cs (hunk 14 above)
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/Pages/HomePageNonAuthorizedUser.cs:5]: added

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/UserMenuDropDownListPageComponent.cs (hunk 1 above)
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/UserMenuDropDownListPageComponent.cs:1]: done

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/NavigationMenuPageComponent.cs (hunk 3 above)
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/NavigationMenuPageComponent.cs]: removed instances of WebDriverWait class from all pages

**Review by LoveMyDaughter — APPROVED** [review]

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs (hunk 2 above)
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs]: done

Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/PageComponents/NavigationMenuPageComponents/SignInSignUpHeaderButtons.cs]: done

**Review by LoveMyDaughter — APPROVED** [review]
Changed a lot.

**Review by dmatsev — COMMENTED** [review]
Well done!

**Pull request merged**